### PR TITLE
Adding fix for nonetype error in global credentials

### PR DESCRIPTION
**Pull request by abimishr**
1.  **Git PR Link:**https://github.com/madhansansel/dnacenter-ansible/pull/190
2.  **Problem Description**: User sees a typeError when a particular global credential is not present and tries to use the same to create a discovery
3.  **Root Cause:** This happens due to not handling None type error properly.
4.  **Code Changes**: Have added a check in the handle_global_credentials API.
5. **Testing and Automation**:  For testing we tried to discover without any valid netconf port in the Global credentials. There was a failure saying 'No Global Net Conf Ports are present in the Cisco Catalyst Center'.

(pyats_env) admin@en-soln-lnx:~$ ansible-playbook /home/admin/madhan_ansible/collections/ansible_collections/cisco/dnac/playbooks/discovery_workflow_manager.yml -vvv --extra-vars "CLUSTERFILE=/home/admin/madhan_ansible/collections/ansible_collections/cisco/dnac/playbooks/cluster.yml"
ansible-playbook [core 2.13.10]
  config file = None
  configured module search path = ['/home/admin/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /home/admin/pyats_env/lib/python3.8/site-packages/ansible
  ansible collection location = /home/admin/madhan_ansible/collections/ansible_collections/cisco/dnac
  executable location = /home/admin/pyats_env/bin/ansible-playbook
  python version = 3.8.10 (default, Nov 22 2023, 10:22:35) [GCC 9.4.0]
  jinja version = 3.1.2
  libyaml = True
No config file found; using defaults
host_list declined parsing /etc/ansible/hosts as it did not pass its verify_file() method
Skipping due to inventory source not existing or not being readable by the current user
script declined parsing /etc/ansible/hosts as it did not pass its verify_file() method
auto declined parsing /etc/ansible/hosts as it did not pass its verify_file() method
Skipping due to inventory source not existing or not being readable by the current user
yaml declined parsing /etc/ansible/hosts as it did not pass its verify_file() method
Skipping due to inventory source not existing or not being readable by the current user
ini declined parsing /etc/ansible/hosts as it did not pass its verify_file() method
Skipping due to inventory source not existing or not being readable by the current user
toml declined parsing /etc/ansible/hosts as it did not pass its verify_file() method
[WARNING]: No inventory was parsed, only implicit localhost is available
[WARNING]: provided hosts list is empty, only localhost is available. Note that the implicit localhost does not match 'all'
[WARNING]: running playbook inside collection cisco.dnac
Skipping callback 'default', as we already have a stdout callback.
Skipping callback 'minimal', as we already have a stdout callback.
Skipping callback 'oneline', as we already have a stdout callback.

PLAYBOOK: discovery_workflow_manager.yml *******************************************************************************************************************************************************
1 plays in /home/admin/madhan_ansible/collections/ansible_collections/cisco/dnac/playbooks/discovery_workflow_manager.yml
Read vars_file '{{ CLUSTERFILE }}'
Read vars_file '{{ CLUSTERFILE }}'
Read vars_file '{{ CLUSTERFILE }}'

PLAY [Discover devices using multiple discovery specific credentials and delete all the discoveries] *******************************************************************************************
Read vars_file '{{ CLUSTERFILE }}'
META: ran handlers
Read vars_file '{{ CLUSTERFILE }}'

TASK [Execute discovery devices using MULTI RANGE with various global credentials] *************************************************************************************************************
task path: /home/admin/madhan_ansible/collections/ansible_collections/cisco/dnac/playbooks/discovery_workflow_manager.yml:201
<127.0.0.1> ESTABLISH LOCAL CONNECTION FOR USER: admin
<127.0.0.1> EXEC /bin/sh -c 'echo ~admin && sleep 0'
<127.0.0.1> EXEC /bin/sh -c '( umask 77 && mkdir -p "` echo /home/admin/.ansible/tmp `"&& mkdir "` echo /home/admin/.ansible/tmp/ansible-tmp-1710238655.3555555-337334-6765485194140 `" && echo ansible-tmp-1710238655.3555555-337334-6765485194140="` echo /home/admin/.ansible/tmp/ansible-tmp-1710238655.3555555-337334-6765485194140 `" ) && sleep 0'
Using module file /home/admin/madhan_ansible/collections/ansible_collections/cisco/dnac/plugins/modules/discovery_workflow_manager.py
<127.0.0.1> PUT /home/admin/.ansible/tmp/ansible-local-336751khb5hdws/tmpb93btqfy TO /home/admin/.ansible/tmp/ansible-tmp-1710238655.3555555-337334-6765485194140/AnsiballZ_discovery_workflow_manager.py
<127.0.0.1> EXEC /bin/sh -c 'chmod u+x /home/admin/.ansible/tmp/ansible-tmp-1710238655.3555555-337334-6765485194140/ /home/admin/.ansible/tmp/ansible-tmp-1710238655.3555555-337334-6765485194140/AnsiballZ_discovery_workflow_manager.py && sleep 0'
<127.0.0.1> EXEC /bin/sh -c '/home/admin/pyats_env/bin/python3 /home/admin/.ansible/tmp/ansible-tmp-1710238655.3555555-337334-6765485194140/AnsiballZ_discovery_workflow_manager.py && sleep 0'
<127.0.0.1> EXEC /bin/sh -c 'rm -f -r /home/admin/.ansible/tmp/ansible-tmp-1710238655.3555555-337334-6765485194140/ > /dev/null 2>&1 && sleep 0'
fatal: [localhost]: FAILED! => {
    "changed": false,
    "invocation": {
        "module_args": {
            "config": [
                {
                    "discovery_name": "Multi_global",
                    "discovery_type": "MULTI RANGE",
                    "global_credentials": {
                        "cli_credentials_list": [
                            {
                                "description": "ISE",
                                "username": "cisco"
                            },
                            {
                                "description": "CLI1234",
                                "username": "cli"
                            }
                        ],
                        "http_read_credential_list": [
                            {
                                "description": "HTTP Read",
                                "username": "HTTP_Read"
                            }
                        ],
                        "net_conf_port_list": [
                            {
                                "description": "For_test"
                            }
                        ],
                        "snmp_v3_credential_list": [
                            {
                                "description": "snmpV3",
                                "username": "snmpV3"
                            }
                        ]
                    },
                    "ip_address_list": [
                        "204.1.2.1-204.1.2.5",
                        "204.192.3.40"
                    ],
                    "protocol_order": "ssh"
                }
            ],
            "config_verify": true,
            "dnac_debug": true,
            "dnac_host": "10.22.40.214",
            "dnac_log": true,
            "dnac_log_append": true,
            "dnac_log_file_path": "dnac.log",
            "dnac_log_level": "DEBUG",
            "dnac_password": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER",
            "dnac_port": "443",
            "dnac_username": "admin",
            "dnac_verify": false,
            "dnac_version": "2.3.5.3",
            "state": "merged",
            "validate_response_schema": true
        }
    },
    "msg": "Global netconf ports are not present in the Cisco Catalyst Center"
}

PLAY RECAP *************************************************************************************************************************************************************************************
localhost                  : ok=0    changed=0    unreachable=0    failed=1    skipped=0    rescued=0    ignored=0

### CODE REVIEW

File: plugins/modules/discovery_intent.py [hunk 1]
```diff
@@ -690,6 +756,9 @@ def handle_global_credentials(self, response=None):
             if not isinstance(cli_credentials_list, list):
                 msg = "Global CLI credentials must be passed as a list"
                 self.discovery_specific_cred_failure(msg=msg)
+            if response.get("cliCredential") is None:
+                msg = "No Global CLI credentials are present in the Cisco Catalyst Center"
```
Comment (madhansansel) [inline plugins/modules/discovery_intent.py:760]: msg = "Global CLI credentials are not present in the Cisco Catalyst Center"

File: plugins/modules/discovery_intent.py [hunk 2]
```diff
@@ -762,7 +840,7 @@ def handle_global_credentials(self, response=None):
                                 global_credentials_all["snmpV2cRead"].append(snmp.get("id"))
                         global_credentials_all["snmpV2cRead"] = global_credentials_all["snmpV2cRead"][:cred_len]
                     else:
-                        msg = "Kindly ensure you include both the description and the username for the Global SNMPV2 Read \
+                        msg = "Kindly ensure you include the description for the Global SNMPV2 Read \
```
Comment (madhansansel) [inline plugins/modules/discovery_intent.py:843]: SNMPv2

File: plugins/modules/discovery_intent.py [hunk 3]
```diff
@@ -771,6 +849,9 @@ def handle_global_credentials(self, response=None):
             if not isinstance(snmp_v2_write_credential_list, list):
                 msg = "Global SNMPV2 write credentials must be passed as a list"
                 self.discovery_specific_cred_failure(msg=msg)
+            if response.get("snmpV2cWrite") is None:
+                msg = "No Global SNMPV2 write credentials are present in the Cisco Catalyst Center"
```
Comment (madhansansel) [inline plugins/modules/discovery_intent.py:853]: SNMPv2

File: plugins/modules/discovery_intent.py [hunk 4]
```diff
@@ -812,6 +896,9 @@ def handle_global_credentials(self, response=None):
             if not isinstance(net_conf_port_list, list):
                 msg = "Global net Conf Ports be passed as a list"
                 self.discovery_specific_cred_failure(msg=msg)
+            if response.get("netconfCredential") is None:
+                msg = "No Global Net Conf Ports are present in the Cisco Catalyst Center"
```
Comment (madhansansel) [inline plugins/modules/discovery_intent.py:900]: Global netconf ports are not present ...

File: plugins/modules/discovery_intent.py (hunk 2 above)
Comment (abimishr) [reply plugins/modules/discovery_intent.py]: Addressed

File: plugins/modules/discovery_intent.py (hunk 3 above)
Comment (abimishr) [reply plugins/modules/discovery_intent.py]: Addressed

File: plugins/modules/discovery_intent.py (hunk 4 above)
Comment (abimishr) [reply plugins/modules/discovery_intent.py]: Addressed

**Review by madhansansel — APPROVED** [review]

**Pull request merged**